network-api/networkapi/wagtailpages/pagemodels/blog/blog_index.py

```diff
@@ -407,15 +409,34 @@ def blog_author_detail(self, request: "HttpRequest", profile_slug: str, *args, *
             the profile_slug queried.
         """
         author_profile = get_object_or_404(Profile, slug=profile_slug)
+        authors_frequent_topics = self.get_authors_frequent_topics(author_profile=author_profile)
         return self.render(
             request,
             context_overrides={
                 "author": author_profile,
                 "page": self,
+                "frequent_topics": authors_frequent_topics,
             },
             template="wagtailpages/blog_author_detail_page.html",
         )
 
+    def get_authors_frequent_topics(self, author_profile):
+        BlogPage = apps.get_model("wagtailpages.BlogPage")
+
+        # Retrieve all BlogPages authored by this profile.
+        authored_blog_pages = localize_queryset(BlogPage.objects.filter(authors__author=author_profile))
+
+        # From the previous QS, get the related BlogPageTopic objects.
+        frequent_topics = BlogPageTopic.objects.filter(blogpage__in=authored_blog_pages)
+
+        # Calculate the count of each topic using Counter.
+        topic_counts = Counter(frequent_topics)
```

Comment:
I feel like we can defer all of this to the db in one or two queries (excluding the localisation part). Maybe:

```python
top_topics = BlogPageTopic.objects.filter(blogpage__authors__author=author_profile).annotate(count=Count("name")).order_by("count")[:3]

top_topics = localize_queryset(top_topics, order_by="name")
```
I've tested it with a local copy of prod's db and it seems to work, but I haven't done extensive testing.

In any case, it would be good to have backend tests for this function before we merge.

Reply:
Great suggestion Jhonatan! I have updated the code to use your suggested query, I think it looks a lot cleaner and is easier to understand. 

Working on adding tests now 👍

Reply:
Hi Jhonatan! Wanted to let you know that your query is working great, the only difference is that I had to update it with `order_by("-count")` and remove the `order_by="name"` from localize queryset 🙌 

Thanks!!